irohad/validation/impl/stateful_validator_impl.cpp

```diff
@@ -247,19 +247,15 @@ namespace iroha {
       auto transactions_errors_log = validation::TransactionsErrors{};
       auto valid_proto_txs = validateTransactions(
           proposal.transactions(), temporaryWsv, transactions_errors_log, log_);
-      auto validated_proposal = shared_model::proto::ProposalBuilder()
-                                    .createdTime(proposal.createdTime())
-                                    .height(proposal.height())
-                                    .transactions(valid_proto_txs)
-                                    .createdTime(proposal.createdTime())
-                                    .build();
 
-      log_->info("transactions in verified proposal: {}",
-                 validated_proposal.transactions().size());
-      return std::make_pair(std::make_shared<decltype(validated_proposal)>(
-                                validated_proposal.getTransport()),
+      // Since proposal came from ordering gate it was already validated.
+      // All transactions has been validated aswell
+      // This allows for unsafe construction of proposal
```

Comment:
Is the "for" needed in this line?

Reply:
https://www.merriam-webster.com/dictionary/allow%20for

Please look at third definition:

_to make (something) possible The design of the system allows for easy upgrades._

Considerations in the beginning of the comment make it possible to construct proposal unsafely.